### DISCUSSION TITLE
Git config not working

### DISCUSSION THREAD

**Discussion by wolfenstein11x** [community/community#22483] [New to GitHub]
<p>I have used git and github on this computer I am currently using (Windows 10), many times before, but I had not used it for a couple months. When I tried to make a commit to a new repository yesterday, it prompted me to do the git config --global user.email, and when I enter that, I get this error:</p>
<p>error: could not lock config file C:/ProgramData/SPB_Data/.gitconfig: No such file or directory</p>
<p>I followed that path and found there is no SPB_Data folder there (including hidden files). Also I found that my .gitconfig file is located in C:/Users/cwolf. I don’t know if solving the problem is as simple as making a SPB_Data folder and putting that .gitconfig file in it, but I don’t want to try that if I don’t know what I’m doing.</p>
<p>My question is what do I need to do to get the git config working again, and why did this happen if everything used to be working a couple months ago, same computer, same operating system?</p>

**Comment by darkRaspberry** [comment]
<p>As you stated you are using Windows 10. It’s look like your .gitconfig is broken.<br>
This is minimum .gitconfig in your home folder (In your case <code>C:/Users/cwolf</code>) If the <code>.gitconfig</code> is not in your home folder then rectify it.<br>
This is the global config which is necessary. Try opening it and see do this file look like this.</p>
<pre><code class="lang-auto">[user]
	email = youremail@example.com
	name = Your sweet name
	signingkey = XX26XXXXXXXXXX87
[gpg]
	program = %PATH_OF_GPG
[commit]
	gpgsign = true
</code></pre>
<p>If still it doesn’t work, THEN<br>
Try setting global config through command line by- <code>git config --global user.email "MY_NAME@example.com"</code> It automatically set the <code>.gitconfig</code> in the directory it needed.<br>
You can follow <a href="https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup" rel="noopener nofollow ugc">Offical DOCS</a> to set <code>.gitconfig</code>.</p>
<aside class="quote no-group" data-username="wolfenstein11x" data-post="1" data-topic="145512">
<div class="title">
<div class="quote-controls"></div>
 wolfenstein11x:</div>
<blockquote>
<p>My question is what do I need to do to get the git config working again, and why did this happen if everything used to be working a couple months ago, same computer, same operating system?</p>
</blockquote>
</aside>
<p>This MAY happen because you updated your git/system and something unexpected happen or any app misbehaved with your git.<br>
IF you can then try a clean and updated git install. (Uninstall git and install it from offical website. <a href="https://git-scm.com/" rel="noopener nofollow ugc">Git</a>)<br>
If nothing happens then I can’t help.</p>

**Comment by wolfenstein11x** [comment]
<p>This is what my .gitconfig file looks like, located in C:\Users\cwolf:</p>
<p>[user]<br>
name = Charles<br>
email = <a href="mailto:c.wolfe132@gmail.com">c.wolfe132@gmail.com</a><br>
[core]<br>
editor = ‘C:/Program Files (x86)/Notepad++/notepad++.exe’<br>
autocrlf = true<br>
excludesfile = C:\Users\cwolf\Documents\gitignore_global.txt<br>
[filter “lfs”]<br>
smudge = git-lfs smudge – %f<br>
process = git-lfs filter-process<br>
required = true<br>
clean = git-lfs clean – %f</p>
<p>And the file in C:\ProgramData\Git just called “config” looks like this:</p>
<p>[core]<br>
symlinks = false<br>
autocrlf = true<br>
fscache = true<br>
[color]<br>
diff = auto<br>
status = auto<br>
branch = auto<br>
interactive = true<br>
[help]<br>
format = html<br>
[rebase]<br>
autosquash = true</p>
<p>I am going to try uninstalling and reinstalling Git now, and I hopefully that works</p>

**Comment by darkRaspberry** [comment]
<aside class="quote no-group" data-username="darkRaspberry" data-post="2" data-topic="145512">
<div class="title">
<div class="quote-controls"></div>
 darkRaspberry:</div>
<blockquote>
<p>Try setting global config through command line by- <code>git config --global user.email "MY_NAME@example.com"</code> It automatically set the <code>.gitconfig</code> in the directory it needed.</p>
</blockquote>
</aside>
<p>Did u try this. If <code>.gitconfig</code> file location changed then it will create it at required place.<br>
Else Best of luck!</p>

**Comment by wolfenstein11x** [comment]
<p>I just uninstalled Git and then reinstalled it, and tried a git config again, same error:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/7/37c1c1a8fbfff81e015814f63b57803963883537.png" data-download-href="/uploads/short-url/7XfnHhOqlqk5oqTRtAzC1rYFAX5.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/20327817/181071388-332e3805-460f-4632-8380-ec0eb4d28511.png" alt="image" data-base62-sha1="7XfnHhOqlqk5oqTRtAzC1rYFAX5" width="690" height="139" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/7/37c1c1a8fbfff81e015814f63b57803963883537_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">695×141 8.23 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by darkRaspberry** [comment]
<p>Maybe some environment variables have been changed on your system.<br>
I would check the values of <code>HOME</code> , <code>HOMEPATH</code> and <code>USERPROFILE</code> .</p>
<p>see is there any changed then return those to default</p>

**Reply by sweco-seirel** [reply]
@wolfenstein11x @darkRaspberry 
Hi! I encountered the same issue yesterday. The only difference was that I have just installed Git and that my HOME variable was on C/users/SEIREL all the time before and after git installation. I crutch-fixed that with creating a SEIREL directory at C:/Users/SEIREL/AppData/Local/Programs/Git/ where it was claiming to trying to find a config file and where of course it wasn't. So I copied a .gitconfig file into it from my home-directory (C/users/SEIREL) and it worked somehow! Funny thing is that now when I'm changing my configuration, for example changing my global name or email, the changes are seen in C/Users/SEIREL/.gitconfig but not where it used to try to find it!

**Comment by wolfenstein11x** [comment]
<p>Ok I just straight up made a new folder called SPB_Data and copy and pasted my .gitconfig file into it, and that worked.</p>

**Comment by darkRaspberry** [comment]
<p>Yup but just check the enviroment variable. The git is considering home as <code>C:/ProgramData/SPB_Data/</code> but we both know it not.<br>
Home is <code>C:/Users/cwolf</code> in your case.</p>

**Comment by wolfenstein11x** [comment]
<p>USERPROFILE=C:\Users\cwolf<br>
HOMEPATH=\Users\cwolf</p>
<p>but then</p>
<p>HOME=C:\ProgramData\SPB_Data</p>
<p>Ok just to make sure I don’t accidentally break something, what’s the best way to reset that one environment variable</p>

**Comment by darkRaspberry** [comment, marked as answer]
<p>Just edit it. This is the culprit.<br>
<strong>May be</strong> you installed some app in recent day. And you given admin privilege and the app changed the path. Just change it and it will set everything good for any app that depend on home directory for config like git or node.</p>
<p>NOOB HINT- Keep a eye on slash (<code>/</code> ‘’) both are different.</p>

**Comment by wolfenstein11x** [comment]
<p>Thank you for your help. You guys are way more chill than the people on stack overflow</p>

**Comment by darkRaspberry** [comment]
<p>Did you tried changing the environment variable? Did it worked?</p>
<p>Mark this question as solved if satisfied. Or drop a like.</p>

**Comment by naickercuan** [comment]
<p>Had the same problem on Win10. Solved by changing windows env variable ‘HOME’ to<br>
’ C:\Users\ \ ’ .</p>
<p>Seems as though git usings this env variable as its base directory</p>

**Comment by AysEngineer** [comment]
<p>Hi, there had been so much time, but I wanted to solved your problem. I opened SPB_Data File where git was setted, like that  the problem was solved. By the way,  seen .git file in SPB_Data file when I opened this.</p>
<p>Best regards! See you.</p>

**Comment by RalphLH** [comment]
Bom dia!

error: no action specified ao dar o comando git config. O que pode esta acontecendo?
